src/java.xml.crypto/share/classes/javax/xml/crypto/dsig/spec/RSAPSSParameterSpec.java

```diff
@@ -0,0 +1,145 @@
+ *    &lt;xs:complexType name="MaskGenerationFunctionType"&gt;
+ *        &lt;xs:sequence&gt;
+ *            &lt;xs:element ref="ds:DigestMethod" minOccurs="0"/&gt;
+ *        &lt;/xs:sequence&gt;
+ *        &lt;xs:attribute name="Algorithm" type="xs:anyURI"
+ *           default="http://www.w3.org/2007/05/xmldsig-more#MGF1"/&gt;
+ *    &lt;/xs:complexType&gt;
+ * </code></pre>
+ *
+ * @implSpec
+ * If the parameter is not specified when using the RSASSA-PSS signature
+ * algorithm, the default parameter is used, which uses SHA-256 as the
+ * {@code DigestMethod}, MGF1 with SHA-256 as the
+ * {@code MaskGenerationFunction}, 32 as {@code SaltLength}, and 1 as
+ * {@code TrailerField}. This is equivalent to the parameter-less signature
+ * method as defined by http://www.w3.org/2007/05/xmldsig-more#sha256-rsa-MGF1.
+ *
+ * @since 17
+ * @see SignatureMethod
+ * @see <a href="https://www.ietf.org/rfc/rfc6931.txt">RFC 6931</a>
+ */
+public final class RSAPSSParameterSpec implements SignatureMethodParameterSpec {
+
+    private final PSSParameterSpec spec;
+
+    /**
+     * Creates a new {@code RSAPSSParameterSpec} object with the specified
+     * {@link PSSParameterSpec} object encapsulated.
+     *
+     * @param spec the input {@code PSSParameterSpec} to be encapsulated
+     */
+    public RSAPSSParameterSpec(PSSParameterSpec spec) {
+        this.spec = Objects.requireNonNull(spec);
+    }
+
+    /**
+     * Returns the encapsulated {@code PSSParameterSpec} object.
+     *
+     * @return the encapsulated {@code PSSParameterSpec} object
+     */
+    public PSSParameterSpec getPSSParameterSpec() {
+        return spec;
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(
+                spec.getDigestAlgorithm(),
+                spec.getSaltLength(),
+                spec.getTrailerField(),
+                mgfParamsAsString(spec));
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (!(obj instanceof RSAPSSParameterSpec)) {
+            return false;
+        }
+        PSSParameterSpec other = ((RSAPSSParameterSpec)obj).spec;
+        return Objects.equals(spec.getDigestAlgorithm(), other.getDigestAlgorithm())
+                && Objects.equals(mgfParamsAsString(spec), mgfParamsAsString(other))
+                && spec.getSaltLength() == other.getSaltLength()
+                && spec.getTrailerField() == other.getTrailerField();
+    }
+
+    private static String mgfParamsAsString(PSSParameterSpec spec) {
+        String output = spec.getMGFAlgorithm() + ":";
+        AlgorithmParameterSpec mgfSpec = spec.getMGFParameters();
+        if (mgfSpec instanceof MGF1ParameterSpec) {
+            output += ((MGF1ParameterSpec)mgfSpec).getDigestAlgorithm();
+        } else {
+            output += System.identityHashCode(mgfSpec);
+        }
+        return output;
+    }
+
+    @Override
+    public String toString() {
```

Comment:
Add specification.

Reply:
OK.

Reply:
Actually, on second thought, I don't think we should override `toString`. None of the other spec classes override `toString`.

Reply:
I'll remove it.